common-client/src/main/java/com/dell/cpsd/service/common/client/ssh/SSHClient.java

```
@@ -236,6 +236,30 @@ public void copyFile(String localPath, String remotePath) throws IOException
         }
     }
 
+    /**
+     *
+     * @param remotePath Directory to create on target machine
+     *                   Note this method assumes ownership of target directory by executor
+     * @throws IOException
+     */
+    public void createRemoteDirectory(final String remotePath) throws IOException
+    {
+        String mkdir = "mkdir -p %s";
+        SSHCommandResponse responseMkdir = null;
+        try
+        {
+            responseMkdir = executeCommand(String.format(mkdir, remotePath));
+            if (responseMkdir.getExitCode() != 0)
+            {
+                throw new IOException(String.format("Failed to create create remote directory %s", responseMkdir.getError()));
```

Won't this exception be eaten by the catch() below?

---

Yup I was just going to grab the exit code and fire it on as an exception in this case as returning an error probably didn't make much sense to anything using this